robyn/openapi.py

```diff
@@ -0,0 +1,267 @@
+
+        if modified_endpoint not in self.openapi_schema["paths"]:
+            self.openapi_schema["paths"][modified_endpoint] = {}
+        self.openapi_schema["paths"][modified_endpoint][route_type] = path_obj
+
+    def add_subrouter_paths(self, subrouter_openapi):
+        """
+
+        adds the subrouter paths to main router's openapi specs
+
+        :param subrouter_openapi: the OpenAPI object of the current subrouter
+        """
+        paths = subrouter_openapi.openapi_schema["paths"]
+
+        for path in paths:
+            self.openapi_schema["paths"][path] = paths[path]
+
+    def get_path_obj(self, endpoint: str, summary: str, tags: list, query_params: TypedDict = None):
+        """
+        get the "path" openapi object according to spec
+
+        :param endpoint: the endpoint to be added
+        :param summary: short summary of the endpoint (to be fetched from the endpoint defenition by default)
+        :param tags: tags -- for grouping of endpoints
+        :param query_params: query params for the function
+
+        :return: the "path" openapi object according to spec
+        """
+        modified_endpoint = endpoint
+
+        path_params = endpoint.split(":")
+        parameters = []
+
+        if len(path_params) > 1:
+            path = path_params[0]
+
+            modified_endpoint = path[:-1] if path.endswith("/") else path
+
+            for param in path_params[1:]:
+                param_name = param[:-1] if param.endswith("/") else param
+
+                parameters.append(
+                    {
+                        "name": param_name,
+                        "in": "path",
+                        "required": True,  # should handle this too.
+                        "schema": {"type": "string"},
+                    }
+                )
+                modified_endpoint += "/{" + param_name + "}"
+
+        if query_params:
+            for query_param in query_params.__annotations__:
+                # ugly hack -- should fix!
+                param_type = re.findall("'[a-z]+'", str(query_params.__annotations__[query_param]))[0].replace("'", "")
+
+                parameters.append(
+                    {
+                        "name": query_param,
+                        "in": "query",
+                        "required": False,
+                        "schema": {"type": param_type},
+                    }
+                )
+
+        return modified_endpoint, {
+            "summary": summary,
+            "tags": tags,
+            "parameters": parameters,
+            "responses": {
+                "200": {
+                    "description": "Successful Response",
+                }
+            },
+        }
+
+    def docs_handler(self):
+        """
+        get the swagger html page
+        @return: the swagger html page
+        """
+        json.dumps(self.openapi_schema)
```

Comment:
not needed imo

Reply:
It is needed! The openapi.json file won't get generated automatically -- the swagger page will raise an error then.

Reply:
mention it in docstring.